### PR TITLE
fix: memory leak in `Import`

### PR DESCRIPTION
**Pull request by yihuang**
Closes: #772 

don't need to reference the children, which recursively retain the whole subtree in memory.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,10 @@
 - [#718](https://github.com/cosmos/iavl/pull/718) Fix `traverseNodes` unexpected behaviour
 - [#770](https://github.com/cosmos/iavl/pull/770) Add `WorkingVersion()int64` API.
 
+### Bug Fixes
+
+- [#]() Fix memory leak in `Import`.
```
Comment (yihuang) [inline CHANGELOG.md:15]:
```suggestion
- [#773](https://github.com/cosmos/iavl/pull/773) Fix memory leak in `Import`.
```

File: nodedb.go [hunk 2]
```diff
@@ -55,7 +55,7 @@ var (
 	metadataKeyFormat = keyformat.NewKeyFormat('m', 0) // m<keystring>
 )
 
-var errInvalidFastStorageVersion = fmt.Errorf("Fast storage version must be in the format <storage version>%s<latest fast cache version>", fastStorageVersionDelimiter)
+var errInvalidFastStorageVersion = fmt.Errorf("fast storage version must be in the format <storage version>%s<latest fast cache version>", fastStorageVersionDelimiter)
```
Comment (joshlopes) [inline nodedb.go:58]: Just curious why this change

File: import.go [hunk 3]
```diff
@@ -133,19 +133,28 @@ func (i *Importer) Add(exportNode *ExportNode) error {
...
-		node.rightNodeKey = node.rightNode.nodeKey
-		node.size = node.leftNode.size + node.rightNode.size
+		leftNode := i.stack[stackSize-2]
+		rightNode := i.stack[stackSize-1]
+
+		node.leftNode = leftNode
+		node.rightNode = rightNode
+		node.leftNodeKey = leftNode.nodeKey
+		node.rightNodeKey = rightNode.nodeKey
+		node.size = leftNode.size + rightNode.size
```
Comment (joshlopes) [inline import.go:143]: It seems to me that you moved leftNode and rightNode to vars and re-used, but isn't that the same has `node.leftNode` and `node.rightNode` or does GO have an issue with using the object?

File: import.go [hunk 4]
```diff
@@ -133,19 +133,28 @@ func (i *Importer) Add(exportNode *ExportNode) error {
...
+		if err := i.writeNode(rightNode); err != nil {
 			return err
 		}
 		i.stack = i.stack[:stackSize-2]
+
+		// remove the recursive references to avoid memory leak
+		leftNode.leftNode = nil
+		leftNode.rightNode = nil
+		rightNode.leftNode = nil
+		rightNode.rightNode = nil
```
Comment (joshlopes) [inline import.go:157]: Guess this is the actual fix?

**Comment by joshlopes** [comment]
@tac0turtle why are you removing comments ? 👎

**Comment by tac0turtle** [comment]
removing what comments? i didnt touch anything other than merge

File: nodedb.go (hunk 2 above)
Comment (yihuang) [reply nodedb.go:58]: There a lint warns about the leading upper case in string

File: import.go (hunk 3 above)
Comment (yihuang) [reply import.go:143]: Yes, it's equivalent, it's just I think it makes code shorter this way, because the node.leftNode will be repeated many times

File: import.go (hunk 4 above)
Comment (yihuang) [reply import.go:157]: Yes

**Comment by joshlopes** [comment]
@tac0turtle the comments are back, probably a hiccup within GH API, basically I left the comments above and you merged the PR - I saw the event merged and the comments were no were so I assumed you had deleted them to merge/don't stale. 

Apologies! 🙇
